llvm/lib/Support/APFloat.cpp

```diff
@@ -396,6 +404,8 @@ static inline Error createError(const Twine &Err) {
 }
 
 static constexpr inline unsigned int partCountForBits(unsigned int bits) {
+  if (bits == 0)
+    return 1;
```

Comment:
Can avoid reaching here?

Reply:
I tried that before and it resulted in extra checks in other places.
However, I have used std::max(1, ...) now, which avoids the specialized if-check.